### PR TITLE
Add compile test with clang-tidy header filter.

### PR DESCRIPTION
**Pull request by SansoneG**

### CODE REVIEW

File: pilz_testutils/CMakeLists.txt [hunk 1]
```diff
@@ -34,7 +34,7 @@ if(CATKIN_ENABLE_CLANG_TIDY)
     message(FATAL_ERROR "clang-tidy not found.")
   else()
     message(STATUS "clang-tidy found: ${CLANG_TIDY_EXE}")
-    set(CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_EXE}")
+    set(CMAKE_CXX_CLANG_TIDY "${CLANG_TIDY_EXE}" "-header-filter=.*")
```
Comment (martiniil) [inline pilz_testutils/CMakeLists.txt]: Would it be equivalent to use `HeaderFilterRegex=".*"` in the `.clang-tidy` file?

Comment (SansoneG) [reply pilz_testutils/CMakeLists.txt]: Should be. I will test it out and put it in the `.clang-tidy`, since that is probably preferred.

Comment (SansoneG) [reply pilz_testutils/CMakeLists.txt]: @martiniil Done

**Comment by agutenkunst** [comment]
@martiniil this PR states no intention. 
`compile_test.cpp` does not contain anything related to pilz_industrial_motion_testutils. So why are changes needed there? Could this be split in multiple PRs?

**Comment by agutenkunst** [comment]
Not sure about **pilz_industrial_motion_testutils** but I think for **pilz_testutils** something like https://github.com/PilzDE/pilz_common/pull/12 is more appropriate.

**Comment by martiniil** [comment]
> 
> 
> @martiniil this PR states no intention.
> `compile_test.cpp` does not contain anything related to pilz_industrial_motion_testutils. So why are changes needed there? Could this be split in multiple PRs?

Yes, we can split it. There are changes needed because the header files were not checked before. By the way this is true for our other packages, too.

**Pull request closed without merging**